### PR TITLE
Convert controller replica files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

**Comment by scottilee** [comment]
Fixed merge conflict and added .structured_logging file.

File: pkg/controller/replicaset/replica_set.go [hunk 1]
```diff
@@ -372,7 +372,7 @@ func (rsc *ReplicaSetController) addPod(obj interface{}) {
 		if err != nil {
 			return
 		}
-		klog.V(4).Infof("Pod %s created: %#v.", pod.Name, pod)
+		klog.V(4).InfoS("Pod created.", "pod", klog.KObj(pod))
```
Comment (yangjunmyfm192085) [inline pkg/controller/replicaset/replica_set.go:375]:
```suggestion
		klog.V(4).InfoS("Pod created", "pod", klog.KObj(pod))
```

File: pkg/controller/replicaset/replica_set.go [hunk 2]
```diff
@@ -386,7 +386,7 @@ func (rsc *ReplicaSetController) addPod(obj interface{}) {
 	if len(rss) == 0 {
 		return
 	}
-	klog.V(4).Infof("Orphan Pod %s created: %#v.", pod.Name, pod)
+	klog.V(4).InfoS("Orphan Pod created.", "pod", klog.KObj(pod))
```
Comment (yangjunmyfm192085) [inline pkg/controller/replicaset/replica_set.go:389]:
```suggestion
	klog.V(4).InfoS("Orphan Pod created", "pod", klog.KObj(pod))
```

File: pkg/controller/replicaset/replica_set.go [hunk 3]
```diff
@@ -435,7 +435,7 @@ func (rsc *ReplicaSetController) updatePod(old, cur interface{}) {
 		if rs == nil {
 			return
 		}
-		klog.V(4).Infof("Pod %s updated, objectMeta %+v -> %+v.", curPod.Name, oldPod.ObjectMeta, curPod.ObjectMeta)
+		klog.V(4).InfoS("Pod updated.", "currPod", klog.KObj(curPod), "prevPod", klog.KObj(oldPod))
```
Comment (yangjunmyfm192085) [inline pkg/controller/replicaset/replica_set.go:438]:
```suggestion
		klog.V(4).InfoS("Pod updated", "currPod", klog.KObj(curPod), "prevPod", klog.KObj(oldPod))
```

File: pkg/controller/replicaset/replica_set.go [hunk 4]
```diff
@@ -460,7 +460,7 @@ func (rsc *ReplicaSetController) updatePod(old, cur interface{}) {
 		if len(rss) == 0 {
 			return
 		}
-		klog.V(4).Infof("Orphan Pod %s updated, objectMeta %+v -> %+v.", curPod.Name, oldPod.ObjectMeta, curPod.ObjectMeta)
+		klog.V(4).InfoS("Orphan Pod updated.", "currPod", klog.KObj(curPod), "prevPodObjectMeta", klog.KObj(oldPod))
```
Comment (yangjunmyfm192085) [inline pkg/controller/replicaset/replica_set.go:463]:
```suggestion
		klog.V(4).InfoS("Orphan Pod updated", "currPod", klog.KObj(curPod), "prevPodObjectMeta", klog.KObj(oldPod))
```

File: pkg/controller/replicaset/replica_set.go [hunk 5]
```diff
@@ -503,7 +503,7 @@ func (rsc *ReplicaSetController) deletePod(obj interface{}) {
 		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %#v: %v", rs, err))
 		return
 	}
-	klog.V(4).Infof("Pod %s/%s deleted through %v, timestamp %+v: %#v.", pod.Namespace, pod.Name, utilruntime.GetCaller(), pod.DeletionTimestamp, pod)
+	klog.V(4).InfoS("Pod deleted through caller.", "pod", klog.KObj(pod), "caller", utilruntime.GetCaller(), "DeletionTimestamp", pod.DeletionTimestamp)
```
Comment (yangjunmyfm192085) [inline pkg/controller/replicaset/replica_set.go:506]:
```suggestion
	klog.V(4).InfoS("Pod deleted through caller", "pod", klog.KObj(pod), "caller", utilruntime.GetCaller(), "DeletionTimestamp", pod.DeletionTimestamp)
```

File: pkg/controller/replicaset/replica_set.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/replicaset/replica_set.go:375]: Fixed.

File: pkg/controller/replicaset/replica_set.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/replicaset/replica_set.go:389]: Fix.ed

File: pkg/controller/replicaset/replica_set.go (hunk 3 above)
Comment (scottilee) [reply pkg/controller/replicaset/replica_set.go:438]: Fixed.

File: pkg/controller/replicaset/replica_set.go (hunk 4 above)
Comment (scottilee) [reply pkg/controller/replicaset/replica_set.go:463]: Fixed.

File: pkg/controller/replicaset/replica_set.go (hunk 5 above)
Comment (scottilee) [reply pkg/controller/replicaset/replica_set.go:506]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 Addressed your comments and fixed the broken test.

**Pull request closed without merging**